src/main/java/uk/ac/ebi/eva/evaseqcol/datasource/NCBISeqColDataSource.java

```diff
@@ -86,4 +87,27 @@ public Optional<Map<String, Object>> getAllPossibleSeqColExtendedData(String acc
 
         return Optional.of(seqColResultData);
     }
+
+    public Optional<Map<String, Object>> getAllPossibleSeqColExtendedData(String insdcAccession, String fastFileContent) throws IOException {
```

Comment:
This being a generic function for any fasta file, I'm not sure it makes sense to put it in `NCBISeqColDataSource.java`